animation_nodes/nodes/geometry/c_utils.pyx

```diff
@@ -20,7 +21,7 @@ cdef intersect_LineLine(Vector3 *firstLineStart, Vector3 *firstLineEnd,
     subVec3(&direction1, firstLineEnd, firstLineStart)
     subVec3(&direction2, secondLineEnd, secondLineStart)
     crossVec3(&normal, &direction1, &direction2)
-    if lengthVec3(&normal) < 1e-6:
+    if lengthVec3(&normal) < 1e-6 or lengthVec3(&direction1) == 0 or lengthVec3(&direction2) == 0:
```

Comment:
is this extra check really necessary? The length of the normal is 0 when one of the directions is 0, right?
Also, if you just want to compare to 0 it would be faster to use the lengthsquared function.
Another good idea would be to create another `isZeroVec3` in the vector math file that just checks the individual components.